pages/deployments/oidc/oidc_with_aws.md

```diff
@@ -0,0 +1,204 @@
+
+- **Provider URL**: `https://agent.buildkite.com`
+
+- **Audience**: `sts.amazonaws.com`
+
+## Step 2: Create a new (or update an existing) IAM role to use with your pipelines
+
+Creating new or updating existing IAM roles is conducted through your AWS account.
+
+Learn more about how to do this in the [Creating a role using custom trust policies (console)](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-custom.html) page of the AWS documentation.
+
+As part of this process:
+
+1. Choose the **Custom trust policy** role type.
+
+1. Copy the following example trust policy in the following JSON code block and paste it into a code editor:
+
+    ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": "arn\:aws\:iam:\:AWS_ACCOUNT_ID\:oidc-provider/agent.buildkite.com"
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                    "StringEquals": {
+                        "agent.buildkite.com:aud": "sts.amazonaws.com"
+                    },
+                    "StringLike": {
+                        "agent.buildkite.com:sub": "organization\:ORGANIZATION_SLUG\:pipeline:PIPELINE_SLUG\:ref\:REF\:commit\:BUILD_COMMIT\:step\:STEP_KEY"
+                    },
+                    "IpAddress": {
+                        "aws:SourceIp": [
+                            "AGENT_PUBLIC_IP_ONE",
+                            "AGENT_PUBLIC_IP_TWO"
+                        ]
+                    }
+
+                }
+            }
+        ]
+    }
+    ```
+
+1. Modify the `Principal` section of the pasted code snippet accordingly:
+    1. Ensure that this is set to `Federated`, and that the `oidc-provider` setting references the **Provider URL** you [configured above](#step-1-set-up-an-oidc-provider-in-your-aws-account) (that is, `agent.buildkite.com`).
+    1. Change `AWS_ACCOUNT_ID` to your actual AWS account ID.
+
+1. Modify the `Condition` section of the code snippet accordingly:
+    1. Ensure the `StringEquals` subsection's _audience_ field name (your provider URL appended by `:aud`—`agent.buildkite.com:aud`) has a value that matches the **Audience** you [configured above](#step-1-set-up-an-oidc-provider-in-your-aws-account) (that is, `sts.amazonaws.com`).
+    1. Ensure the `StringLike` subsection's _subject_ field name (your provider URL appended by `:sub`—`agent.buildkite.com:sub`) has at least one value that matches the format: `organization:ORGANIZATION_SLUG:pipeline:PIPELINE_SLUG:ref:REF:commit:BUILD_COMMIT:step:STEP_KEY`, where the constituent fields of this line determine the conditions (that is, when the values specified in these constituent fields have been met) under which the IAM role is granted in exchange for the OIDC token. The following constituent field's value:
```

Comment:
Oh I love how you were able to rephrase this! Thank you!

Reply:
You're welcome! Thanks for answering all my earlier questions, which certainly helped :-)